src/WebJobs.Script.WebHost/Controllers/HostController.cs

```diff
@@ -109,6 +111,24 @@ public IActionResult Ping([FromServices] IScriptHostManager scriptHostManager)
             return Ok();
         }
 
+        [HttpPost]
+        [Route("admin/host/scale/status")]
+        [Authorize(Policy = PolicyNames.AdminAuthLevelOrInternal)]
+        [ResponseCache(NoStore = true, Location = ResponseCacheLocation.None)]
+        [RequiresRunningHost]
```

Comment:
Because I'm using [FromServices] to pull the scale manager and it relies on host level services, I've applied [RequiresRunningHost]. This will mean 503 errors will be returned to scale controller if the host is offline (app_offline.htm), which the scale controller will handle.

Another option is to dynamically check for the service in the services collection, not using [FromServices]

Reply:
There are also periods where there are no host services during restarts. There's some tiny periods where we have no active host and therefore no host services. That should apply here as well -- just wanted to call it out.

Reply:
Right - the important point is to make sure that the request delays until the services are available. Without this, I was seeing some failures on scale controller requests, when attempting to resolve the FunctionsScaleManager service.